core/controllers/payload_validator.py

```diff
@@ -21,13 +21,18 @@
 from __future__ import absolute_import
 from __future__ import unicode_literals
 
+import python_utils
 import schema_utils
 
-from typing import Any, Dict, List, Text, Tuple # isort:skip  pylint: disable= wrong-import-order, wrong-import-position, unused-import, import-only-modules
+from typing import Any, Dict, List, Text, Tuple, Optional, Union # isort:skip  pylint: disable= wrong-import-order, wrong-import-position, unused-import, import-only-modules
 
+SCHEMA_TYPE_BOOL = schema_utils.SCHEMA_TYPE_BOOL
 
-def validate(handler_args, handler_args_schemas, allowed_extra_args):
-    # type: (Any, Any, bool) -> Tuple[Dict[Any, Any], List[Text]]
+
+def validate(
+        handler_args, handler_args_schemas, allowed_extra_args,
+        allow_string_to_bool_conversion=False):
+    # type: (Any, Any, bool, bool) -> Tuple[Dict[Text, Any], List[Text]]
```

Comment:
Please don't use this old format, add the types directly to the code. e.g.
```suggestion
def validate(
        handler_args: Any,
        handler_args_schemas: Any, 
        allowed_extra_args: bool,
        allow_string_to_bool_conversion: bool = False
) -> Tuple[Dict[Text, Any], List[Text]]:
```

ditto in other places

Reply:
Thank you for guidance, updated.